ext/soap/soap.c

```diff
@@ -1542,11 +1542,7 @@ PHP_METHOD(SoapServer, handle)
 			if (zend_hash_find_ptr_lc(function_table, Z_STR(h->function_name)) != NULL ||
 			    ((service->type == SOAP_CLASS || service->type == SOAP_OBJECT) &&
 			     zend_hash_str_exists(function_table, ZEND_CALL_FUNC_NAME, sizeof(ZEND_CALL_FUNC_NAME)-1))) {
-				if (service->type == SOAP_CLASS || service->type == SOAP_OBJECT) {
-					call_status = call_user_function(NULL, soap_obj, &h->function_name, &h->retval, h->num_params, h->parameters);
-				} else {
-					call_status = call_user_function(EG(function_table), NULL, &h->function_name, &h->retval, h->num_params, h->parameters);
-				}
+				call_status = call_user_function(NULL, soap_obj, &h->function_name, &h->retval, h->num_params, h->parameters);
```

Comment:
Now you're passing soap_obj not only for `service->type == SOAP_CLASS || service->type == SOAP_OBJECT` but now also for `zend_hash_find_ptr_lc(function_table, Z_STR(h->function_name)) != NULL`. So this is changed behaviour, or what am I missing?

Reply:
`soap_obj` is NULL if `service->type != SOAP_CLASS && service->type != SOAP_OBJECT` so the code is equivalent.

(yes the way this is done above is quite convoluted.)

Reply:
I see now, you're right. Looking at the condition now that's quite logical but the code was confusingly structured.